docs/deployment-guide.md

```diff
@@ -213,20 +213,84 @@ If you have a valid license place your `.lic` file in the `configuration_files/l
 
 > NOTE: You can also [upload a license](https://docs.alfresco.com/content-services/latest/admin/license/) via the Admin Console once the system is running.
 
-### Alfresco/Solr authentication
+### Secrets management
 
-As of ACS 7.2 and/or Search services 2.0.3, the repository <--> solr communication requires to be authenticated. The playbook will set up that authentication scheme using the new `secret` method.
-This methods needs to be passed a shared secret. In order to do so use the variable below:
+This playbook expects that security-relevant variables are configured in `vars/secrets.yml` file.
 
-```yaml
-reposearch_shared_secret: dummy
+Is it strongly recommended to enable [Ansible Vault](https://docs.ansible.com/ansible/latest/user_guide/vault.html) in order to avoid keeping secrets in plaintext on the control node file-system.
+
+If you run the playbook without configuring Ansible Vault, it will automatically
+generate those secrets but they will be saved as plaintext.
+
+#### Enable Ansible Vault support
+
+The steps required to enable **Ansible Vault** support are:
+
+* Configure a vault password in a file (e.g. `~/.vault_pass.txt`)
+  * Optionally you can autogenerate a strong password with:
+
+    ```bash
+    openssl rand -base64 21 > ~/.vault_pass.txt
+    ```
+
+* Export the path to the vault password file as `ANSIBLE_VAULT_PASSWORD_FILE` so
+  that can automatically picked-up when running Ansible:
+
+  ```bash
+  export ANSIBLE_VAULT_PASSWORD_FILE=~/.vault_pass.txt
+  ```
+
+Otherwise you can provide the password before each playbook invocation with:
+
+```bash
+ansible-playbook --ask-vault-pass ...
```

Comment:
Maybe worth presenting this way of enabling vaults as the main one (as it's a lot simpler) and the one using vault password file as an automation option (also because storing passwords in file won't be seen as a good default choice)